components/x-gift-article/src/SharedLinkTypeSelector.jsx

```diff
@@ -11,7 +11,9 @@ export const SharedLinkTypeSelector = (props) => {
 		enterpriseRequestAccess,
 		showAdvancedSharingOptions,
 		enterpriseHasCredits,
-		giftCredits
+		giftCredits,
+		hasHighlights,
+		includeHighlights
```

Comment:
**Question:** Could you please show me where these new props come from? I have tried to trace them back but I can't quite find the starting point. As they are used for stuff like `checked` and `value` attributes I am wondering if we need to be defaulting them or type checking them to ensure the values are correct

Reply:
these are not new props , I just moved them from the `AdvancedSharingOptions` to this file since the highlights are no longer exclusive to advance sharing users.

As for where these props come from they are assign when the tool-tip is initialized on next-article
https://github.com/Financial-Times/next-article/blob/42b329f65709ec05984422cc463663d6903e36b6/client/components/gift-article/index.jsx#L125-L143

they also have default values in the initalState in the Gift article component https://github.com/Financial-Times/x-dash/blob/e9062b0215b4593155cdaec985880bf86d67a62f/components/x-gift-article/src/GiftArticle.jsx#L264-L266

Reply:
Thanks